teos/src/api/http.rs

```diff
@@ -300,6 +305,7 @@ pub async fn serve(http_bind: SocketAddr, grpc_bind: String, shutdown_signal: Li
     };
     let (_, server) = warp::serve(router(grpc_conn))
         .bind_with_graceful_shutdown(http_bind, async { shutdown_signal.await });
```

Comment:
I tried replacing the `async { shutdown_signal.await }` with just `shutdown_signal` and worked fine.
Maybe we can simplify this in this PR.

Reply:
Right, `triggered::Listener` actually implements `std::future::Future`, so there is no need to wrap this with async.

Will squeeze that in.

Reply:
Fixed in [bfc5132](https://github.com/talaia-labs/rust-teos/pull/106/commits/bfc5132aeac28b3a3128def4d11dbfe1e0bee50c)